blog/_posts/2019-04-24-MLJ.md

```diff
@@ -0,0 +1,253 @@
+and repeats similar declarations, omitting calls to
+`fit!`. The difference now is that each "variable" (e.g., `Xt`,
+`yhat`) is a node of our pipeline, instead of concrete data:
+
+```julia
+Xs = source(X)
+ys = source(y)
+
+hot = machine(hot_model, Xs)
+Xt = transform(hot, Xs);
+
+tree = machine(tree_model, Xt, ys)
+yhat = predict(tree, Xt)
+```
+
+If we like, we can think of a node as *dynamic data* - "data" because
+it can be called (indexed) on rows, but "dynamic" because the result
+depends on the outcome of training events, which in turn depend on 
+hyperparameter values. For example, after fitting the completed pipeline,
+we can make new predictions like this:
+
+```julia
+fit!(yhat, rows=1:1300)
+ [ Info: Training NodalMachine @ 1…51.
+ [ Info: Spawned 1300 sub-features to one-hot encode feature :Neighborhood.
+ [ Info: Spawned 1300 sub-features to one-hot encode feature :MSSubClass.
+ [ Info: Training NodalMachine @ 1…17.
+ Node @ 1…79 = predict(1…17, transform(1…51, 1…07))
+
+yhat(rows=1301:1302) # to predict on rows of source node
+yhat(Xnew)           # to predict on new data
+156-element Array{Float64,1}:
+ 223956.9999999999 
+ 320142.85714285733
+ ...
+```
+
+
+## Exporting and retraining
+
+Once a pipeline like this has been built and tested on sample data, it
+can be exported as a stand-alone model, ready to be trained on any
+dataset. For details, see the MLJ
+[documentation](https://alan-turing-institute.github.io/MLJ.jl/dev/learning_networks/). In
+the future, Julia macros will allow common architectures (e.g., linear
+pipelines) to be built in a couple of lines.
+
+Finally, we mention that MLJ learning networks, and their exported
+counterparts, are "smart" in the sense that changing a hyperparameter
+does not trigger retraining of component models upstream of the
+change:
+
+```julia
+tree_model.max_depth = 4
+fit!(yhat, rows=1:1300)
+ [ Info: Not retraining NodalMachine @ 1…51. It is up-to-date.
+ [ Info: Updating NodalMachine @ 1…17.
+ Node @ 1…79 = predict(1…17, transform(1…51, 1…07))
+```
+
+
+## Just "Write the math!"
+
+Because of Julia's generic programming features, any kind of operation
+you would normally apply to data (arithmetic, row selection, column
+concatenation, etc) can be overloaded to work with nodes. In this way,
+MLJ's network-building syntax is ecomical, intuitive and easy to
+read. In this respect we have been inspired by [On Machine Learning
+and Programming Languages](https://julialang.org/blog/2017/12/ml&pl).
+
+## Invitation to the community
+We now invite the community, to try out our newly registered packages, [MLJ](https://github.com/alan-turing-institute/MLJ.jl), [MLJModels](https://github.com/alan-turing-institute/MLJModels.jl) and [MLJBase](https://github.com/alan-turing-institute/MLJBase.jl), and provide any feedback or suggestions you may have going forward. We are also particularly interested in hearing how you would use our package, and what features it may be lacking. 
```

Comment:
Do you expect users to directly use MLJBase?

Reply:
ideally no, so removed mention of it